systemtest/src/test/java/io/strimzi/systemtest/security/oauth/OauthAuthorizationST.java

```diff
@@ -260,23 +270,184 @@ void testSuperUserWithOauthAuthorization() {
         ClientUtils.waitForClientSuccess(TEAM_A_CONSUMER_NAME, NAMESPACE, MESSAGE_COUNT);
     }
 
-    @Disabled("Will be implemented in next PR")
+    /**
+     * 1) Try to send messages to topic starting with `x-` with producer from Dev Team A
+     * 2) Change the Oauth listener configuration -> add the maxSecondsWithoutReauthentication set to 30s
+     * 3) Try to send messages with delay of 1000ms (in the meantime, the permissions configuration will be changed)
+     * 4) Get all configuration from the Keycloak (realms, policies) and change the policy so the Dev Team A producer should not be able to send messages to the topic
+     *      starting with `x-` -> updating the policy through the Keycloak API
+     * 5) Wait for the WaitException to appear -> as the producer doesn't have permission for sending messages, the
+     *      job will be in error state
```

Comment:
The question here is in which cases will WaitException occur. Can it occur because the broker is slow or connection with broker is slow? If so the thing will fail due to something else - not due to TopicAuthorizationException. In that case the test will pass even though it didn't properly execute. You have to actually get to the underlying reason for failure, WaitException may not be enough to properly test the expected behaviour here.

But most of all, as far as I can see this test rather than test if reauthentication is activated, assumes that it is activated and properly working, and then tests some expected behavior on top of that assumption. It does not test reauthentication. It tests refreshing of grants under the assumption that reauthentication is active and working as assumed. There may be a bug in how Kafka CR is turned into Kafka Broker `/tmp/strimzi.properties` and even though you activate reauthentication in Kafka CR it may not actually be enabled in Kafka Broker. This test might (depending on the access token timeout) pass even if reauthentication isn't actually activated.

It might even pass (to my understanding, as described above) if the test cluster is simply slow and unresponsive.

I said that testing this was tricky, so it should be worked out in a document before attempting to write a complicated test. This test will probably pass even if reauthentication doesn't work at all.

Reply:
Well you didn't tell me any good approach how to test it until now. I didn't got any information from you.All of this is idea from @see-quick and @Frawless, that we discussed, how I said, the best way how to test it. How I mentioned above, another way how to look if the re-auth is functional is through the log and again, how I mentioned above, we don't want to do it this way.

Also, some of things you described, like testing if the reauth is working properly on Kafka broker etc. is thing of the unit tests or IT, which you should write to make sure, that everything is working as expected. This should not be part of the ST. ST should test some scenario and expect some behaviour.

If you have better idea how to test it, which will not have any limitations and will be simple to test (like you can see in here) go on and ping me.

Reply:
I'll prepare a testing document, how I imagine you could test this functionality. You can always delay starting the work on an issue if you think you don't have enough information from me for example, and are waiting for me to provide clarifications, or a thorough plan or whatever - but do let me know when you need more info from me in order to be able to proceed. In this case I can see that you need it so I'll provide the plan.

Reply:
Okay, let's wait for some better scenario, to have it tested properly. Thanks @mstruk .

Reply:
So should we close this in the meantime?

Reply:
Yeah, maybe we can :)